### PR TITLE
Add erocast extractor.

### LINKED ISSUE #4001: [NSFW] erocast.me
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.05.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Worldwide

### Example URLs

https://erocast.me/track/3212/conversational-audio-6-6-22

### Description

Hi there, I found eroshare.me for audio tracks but it seems yt-dlp doesn't support this site at this time. I'd love if support could be added :) Thanks all!

### Verbose log

```shell
[debug] Command-line config: ['-vU', 'https://erocast.me/track/3212/conversational-audio-6-6-22']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355]
[debug] Python version 3.10.4 (CPython 64bit) - macOS-12.4-arm64-arm-64bit
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[debug] [generic] Extracting URL: https://erocast.me/track/3212/conversational-audio-6-6-22
[generic] conversational-audio-6-6-22: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] conversational-audio-6-6-22: Downloading webpage
[generic] conversational-audio-6-6-22: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://erocast.me/track/3212/conversational-audio-6-6-22
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/yt-dlp/2022.5.18/libexec/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1395, in wrapper
    return func(self, *args, **kwargs)
  File "/opt/homebrew/Cellar/yt-dlp/2022.5.18/libexec/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 1465, in __extract_info
    ie_result = ie.extract(url)
  File "/opt/homebrew/Cellar/yt-dlp/2022.5.18/libexec/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 642, in extract
    ie_result = self._real_extract(url)
  File "/opt/homebrew/Cellar/yt-dlp/2022.5.18/libexec/lib/python3.10/site-packages/yt_dlp/extractor/generic.py", line 4031, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://erocast.me/track/3212/conversational-audio-6-6-22
```

### PR DESCRIPTION
**Pull request by madewokherd**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Adds an extractor for erocast.me.

Fixes #4001

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
📦🚚🎨

Add support for erocast.com videos by importing a new extractor class `ErocastIE` from `erocast.py`.

### Walkthrough
*

</details>

### CODE REVIEW

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/erocast.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            'id': video_id,
+            'title': data.get('title', ''),
+            'formats': formats,
+            'age_limit': 18,
+            'release_timestamp': parse_iso8601(data.get('created_at')),
+            'modified_timestamp': parse_iso8601(data.get('updated_at')),
+            **traverse_obj(data, {
+                'description': 'description',
+                'uploader': ('user', 'name'),
+                'uploader_id': ('user', 'id'),
```
[the remark this answers is not in the document]
Comment (madewokherd) [reply yt_dlp/extractor/erocast.py:65]: Should we do that with all fields expected to be str, or just this one?

[the remark this answers is not in the document]
Comment (bashonly) [reply yt_dlp/extractor/erocast.py:65]: for fields like `description` or `uploader`, you can filter by type with `{str}`

we only want to stringify/transform with `{str_or_none}` when it's an int that needs to be str

**Comment by madewokherd** [comment]
Thanks, updated.

**Review by bashonly — APPROVED** [review]
LGTM. In the future, please avoid force pushing; it makes reviewing more difficult for us, and all commits will be squashed before merge anyways

**Comment by madewokherd** [comment]
Ah sorry, will keep that in mind for next time. Thanks for the review.

File: yt_dlp/extractor/erocast.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        webpage = self._download_webpage(url, video_id)
+
+        data = self._search_json(
+            rf'<script>\s*var song_data_{video_id}\s*=', webpage, 'data', video_id, end_pattern=r'</script>')
+
+        media_url = data.get('file_url') or data['stream_url']
+
+        return {
+            'id': video_id,
+            'formats': self._extract_m3u8_formats(media_url, video_id, 'm4a', m3u8_id='hls'),
```
Comment (Grub4K) [inline yt_dlp/extractor/erocast.py:50]:
```suggestion
        return {
            'id': video_id,
            'formats': self._extract_m3u8_formats(
                data.get('file_url') or data['stream_url'], video_id, 'm4a', m3u8_id='hls'),
```

**Review by Grub4K — CHANGES_REQUESTED** [review]
Additionally some of the whitespace should be condensed

File: yt_dlp/extractor/erocast.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+
+        data = self._search_json(
+            rf'<script>\s*var song_data_{video_id}\s*=', webpage, 'data', video_id, end_pattern=r'</script>')
```
Comment (bashonly) [inline yt_dlp/extractor/erocast.py:44]:
```suggestion
        video_id = self._match_id(url)
        webpage = self._download_webpage(url, video_id)
        data = self._search_json(
            rf'<script>\s*var song_data_{video_id}\s*=', webpage, 'data', video_id, end_pattern=r'</script>')
```

**Review by bashonly — APPROVED** [review]

**Review by Grub4K — APPROVED** [review]

**Pull request merged**